modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/AbstractPageMemoryMvPartitionStorage.java

```diff
@@ -240,9 +249,23 @@ private PageMemoryHashIndexStorage createOrRestoreHashIndex(IndexMeta indexMeta)
             return null;
         }
 
-        throwIfChainBelongsToAnotherTx(versionChain, txId);
+        if (versionChain.isUncommitted()) {
+            UUID chainTxId = versionChain.transactionId();
+
+            assert chainTxId != null;
+
+            throwIfChainBelongsToAnotherTx(versionChain, txId);
+
+            return ReadResult.createFromWriteIntent(
+                    row,
+                    versionChain.transactionId(),
+                    versionChain.commitTableId(),
+                    null,
```

Comment:
It looks weird that there's a gap between table id and partition id, I'd love to see them side by side